specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/stable/2024-06-30/dedicatedhsm.json

```diff
@@ -0,0 +1,762 @@
+          },
+          {
+            "$ref": "#/parameters/DedicatedHsmName"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DedicatedHsm"
+            },
+            "description": "Parameters to create or update the dedicated hsm"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Created dedicated HSM",
+            "schema": {
+              "$ref": "#/definitions/DedicatedHsm"
+            }
+          },
+          "200": {
+            "description": "Created dedicated HSM",
+            "schema": {
+              "$ref": "#/definitions/DedicatedHsm"
+            }
+          },
+          "default": {
+            "description": "The error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/DedicatedHsmError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create a new or update an existing dedicated HSM": {
+            "$ref": "./examples/DedicatedHsm_CreateOrUpdate.json"
+          },
+          "Create a new or update an existing payment HSM": {
+            "$ref": "./examples/PaymentHsm_CreateOrUpdate.json"
+          },
+          "Create a new or update an existing payment HSM with management profile": {
+            "$ref": "./examples/PaymentHsm_CreateOrUpdate_WithManagementProfile.json"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "DedicatedHsms"
+        ],
+        "operationId": "DedicatedHsm_Update",
+        "x-ms-long-running-operation": true,
+        "description": "Update a dedicated HSM in the specified subscription.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DedicatedHsmName"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "x-ms-client-flatten": true,
+            "schema": {
+              "$ref": "#/definitions/DedicatedHsmPatchParameters"
+            },
+            "description": "Parameters to patch the dedicated HSM"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
```

Comment:
>responses [](http://example.com/codeflow?start=9&length=9)

Need a 202 status code for async response.

Reply:
Updated, operation already supports 202 operation.
<img width="1184" alt="image" src="https://github.com/Azure/azure-rest-api-specs/assets/25797422/ab8e56c4-2014-4452-abf1-f19e69938cd4">